loader/test/support/nock-back-plugin.js

```diff
@@ -59,21 +59,49 @@ function setupNockBack(mode, fixturePath) {
  * recordings will be disallowed. Delete the recording files to record new
  * responses if the upstream server has changed.
  *
- * @param {string} name The test name
+ * @param {any} [options] Options for handling recorded data.
+ * @param {boolean} [options.ignoreQuery] Ignore querystrings in requests.
+ * @param {(any) => void} [options.prepareScope] Function that will be called
+ *        with each recorded request definition before installing them to be
+ *        replayed. Use this, for example, to make recordings match when parts
+ *        of a request (e.g. the path) are random or change from test run to
+ *        test run.
  * @param {() => Promise} testFunction A Promise-returning test function
- * @returns {[string, () => Promise]}
+ * @returns {() => Promise}
  */
-function withNockBack(testFunction) {
+function withNockBack(options, testFunction) {
+  if (typeof options === "function") {
+    testFunction = options;
+    options = {};
+  }
+
+  const originalMode = nock.back.currentMode;
   module.exports.setupNockBack();
+  nock.back.setMode("record");
```

Comment:
This feels a little duplicative, so maybe the `setMode` part should come out of `setupNockBack`, or some other solution is needed here.

The issue is that, it turns out being in "record" mode can cause problems for other tests that use normal Nock instead of Nock.Back. My solution here was to turn "record" mode on and off at the start and end of each nock-back test. Not the prettiest but it works.